bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java

```diff
@@ -1209,6 +1209,7 @@ private void writeAndFlush(final Channel channel,
                     }
                 } else {
                     nettyOpLogger.registerFailedEvent(MathUtils.elapsedNanos(startTime), TimeUnit.NANOSECONDS);
+                    errorOut(key);
```

Comment:
> 
> In #4278, the PerChannelBookieClient#addEntryTimeoutNanos is enabled, So you will see that 5 seconds(default timeout) passed after bk1 died, the step 9 trigger PendingAddOp timeout, then cause the issue in #4278.
> 

I have a question, since we have a default request to detect timeout every 5 seconds and remove the timed `CompletionKey` from `completionObjects`, even if the write fails, will the timeout detection task not remove the key?

Reply:
Sure, the timeout detection will remove the key.